### PR TITLE
Add WatchOS 2 support. 

### PR DESCRIPTION
**Pull request by e2l3n**
<b>Description: </b>
Implement support for WatchOS 2. Retarget of https://github.com/Telerik-Verified-Plugins/AppleWatch/pull/18.

<b>Details: </b>
- Rebuilt the `libmmwormhole.a` static lib with latest `MMWormhole` API (version 2.0.0) and split it into two binary files for the different `iOS` and `WatchOS` platforms - `libmmwormhole-ios.a` and  `libmmwormhole-watchos.a` respectively.
- Edit post-hook install scripts in order to add support for `AppBuilder`.
- Address implementation issues after integration with MMWormhole 2.0.0.

<b>TeamPulse: </b>
- [[iOS][AppBuilder][Cordova] Add WatchOS 2 support to AppleWatch plugin.](http://teampulse.telerik.com/view#item/305755)
- [Support for Watch OS 2](http://teampulse.telerik.com/view#item/306737)

<b>Comments:</b> 
The current PR doesn't support or address issues related to the previous implementation of the AppleWatch plugin mainly because WatchOS 2 introduces WatchConnectivity.framework and the watch extension is now located on the watch device where the platform is WatchOS.

### CODE REVIEW

**Comment by e2l3n** [comment]
ping @EddyVerbruggen @galexandrov @mbektchiev 

This PR adds support only to AB. I haven't updated the cordova hooks.

File: src/ios/hooks/ab/add-targets.js [hunk 1]
```diff
@@ -166,55 +175,94 @@ module.exports = function (context) {
 
     var watchKitExtensionResourcesBuildPhase = pbxProject.addBuildPhase([watchKitAppFile.path], 'PBXResourcesBuildPhase', watchKitExtension + ' Resources');
```
Comment (mbektchiev) [inline src/ios/hooks/ab/add-targets.js:176]: Is this phase needed? We don't add it to the new target anymore: https://github.com/Telerik-Verified-Plugins/AppleWatch/pull/19/files#diff-fe59b2a77283faf286762cee9f417ba3R202

File: src/ios/hooks/ab/add-targets.js [hunk 2]
```diff
@@ -166,55 +175,94 @@ module.exports = function (context) {
...
+        value: watchKitExtensionAppexFile.fileRef,
+        comment: watchKitExtensionAppexFileName
+    });
 
-    var watchKitFrameworkBuildPhase = wkext.addFrameworks(pbxProject, watchKitExtension, projectRelativePluginDirPath);
-    var watchKitExtensionNativeTargetGuid = wkext.addTarget(pbxProject,
-        {
+    var watchKitAppPlistJson = plist.parse(fs.readFileSync(watchKitAppPlistFilePath, 'utf8')),
+        watchKitExtensionPlistJson = plist.parse(fs.readFileSync(watchKitExtensionPlistFilePath, 'utf8'));
+    watchKitFrameworkBuildPhase = wkext.addFrameworks(pbxProject, watchKitExtension, projectRelativePluginDirPath);
```
Comment (mbektchiev) [inline src/ios/hooks/ab/add-targets.js:195]: Missing var.

**Comment by mbektchiev** [comment]
:+1: Looks good.

File: src/ios/hooks/ab/add-targets.js [hunk 3]
```diff
@@ -166,55 +175,94 @@ module.exports = function (context) {
...
+    var watchKitExtensionNativeTargetGuid = wkext.addTarget(pbxProject, {
             plistFilePath: watchKitExtensionPlistFilePath,
             displayName: watchKitExtensionDisplayName,
             projectPluginDir: projectRelativePluginDirPath,
             buildPhase: watchKitFrameworkBuildPhase,
             sourcesBuildPhase: watchKitExtensionSourcesBuildPhase,
-            resourcesBuildPhase: watchKitExtensionResourcesBuildPhase,
+            // resourcesBuildPhase: watchKitExtensionResourcesBuildPhase,
```
Comment (galexandrov) [inline src/ios/hooks/ab/add-targets.js:202]: resourcesBuildPhase ?

**Comment by galexandrov** [comment]
:+1:

**Comment by EddyVerbruggen** [comment]
Awesome! Tagging a release and testing in AppBuilder with my WatchOS 2 device. I'll update this issue tomorrow with my findings.

**Comment by EddyVerbruggen** [comment]
@e2l3n I tried building the demo app with v1.0.0 of the plugin but it gives me a warning regarding the deployment target 9.0 which is now set by the plugin (is that really necessary? it may be quite a problem for developers dropping iOS8 support).

The build fails with these errors, can you tell what's going on?

```
    Warning: The Deployment Target constraints ["9.0"] could not be satisfied. Attempting to build with the set 8.2
    Warning: security: unable to open "'/Users/builder/Library/MobileDevice/Provisioning Profiles/d96178e2-b258-462a-998c-969ba24cf00b.mobileprovision'" for reading: No such file or directory
    Error: [xcode-build] Check dependencies: target specifies product type 'com.apple.product-type.watchkit2-extension', but there's no such product type for the 'iphoneos' platform
warning: no rule to process file '/tmp/builds/laJpFnZ3mo2qn6tIxZ/app/Apple_Watch watchkitextension/DetailInterfaceController.h' of type sourcecode.c.h for architecture armv7
warning: no rule to process file '/tmp/builds/laJpFnZ3mo2qn6tIxZ/app/Apple_Watch watchkitextension/GlanceController.h' of type sourcecode.c.h for architecture armv7
```

**Comment by e2l3n** [comment]
@EddyVerbruggen My apologies for the late answer! WatchOS 2.0+ apps require iOS 9.0+ as a minimum deployment target. With WatchOS 2.0 was intruduced a very significant feature - the app extension resides on the watch device which means one is developing against another architecture (armv7k) and uses a brand new WatchOS 2.0 SDK instead of iOS 8.2 + SDK. This is quite a big difference which makes me highly doubtful that we would easily add support to both versions at one place. 

The xcodebuild error  `Error: [xcode-build] Check dependencies: target specifies product type 'com.apple.product-type.watchkit2-extension', but there's no such product type for the 'iphoneos' platform` means that xcodebuild attempts to compile the AppleWatch resources for iPhoneOS which is not possible, it's due to passing `-sdk iphoneos` option to the xcodebuild command. Could you try making another build tomorrow morning because the test environments will be updated with support then?

**Comment by EddyVerbruggen** [comment]
Hey @e2l3n thanks again and don't worry about not responding quicker - we're all super busy!

Can I only try this on test, or will that flag be deployed to production as well (when)?

I understand the changes in WatchOS2, but I'm just wondering how other apps go about this - one would expect to be able to set the WatchOS2 bits to depend on iOS9, but still be able to target the iPhone app to iOS8.

If the user has iOS8 he won't get a Watch app, but if he has iOS9 (about 75% of the iOS users currently) he would also get the watch extension.

In a local Xcode project  I can pull that off by setting the minimum SDK level per target. Is that also possible in AppBuilder, or is it even the way that already works?

Thanks!

**Comment by e2l3n** [comment]
@EddyVerbruggen Eddy, I agree with your thoughts about the minimum deployment target. This PR https://github.com/Telerik-Verified-Plugins/AppleWatch/pull/21 reverts the deployment target value. 

I have addressed several issues with AppBuilder and the new version of the plugin and as soon as these changes are pulled and deployed I will inform you so that you can test the plugin. I'm expecting this to happen in the next day or two.

**Pull request merged**